### PR TITLE
SALTO-4177 Fixed issue layout bugs

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I fixed the issue layout bugs.
1. I Filtered out all the project that aren't company managed, and not of type "software".
2. I removed the owners from the issue layout. 
3. I splitted the filter to three filters - issueLayout, remove porject simplified field, create references for issueLayout. 

---

In a future PR I will resolve the performance problem. 

---

_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts [hunk 1]
```diff
@@ -171,13 +171,15 @@ const filter: FilterCreator = ({ client, config, fetchQuery, getElemIdFunc }) =>
       || !config.fetch.enableIssueLayouts) {
       return
     }
-    const projectToScreenId = await getProjectToScreenMapping(elements)
     const projectIdToProject = Object.fromEntries(
       (await Promise.all(elements.filter(e => e.elemID.typeName === PROJECT_TYPE)
         .filter(isInstanceElement)
+        .filter(project => project.value.simplified === false && project.value.projectTypeKey === 'software')
```
Comment (alonstern) [inline packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:177]: Any reason for `project.value.simplified === false` and not `!project.value.simplified`?

File: packages/jira-adapter/src/config/api_config.ts [hunk 2]
```diff
@@ -838,13 +838,11 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: JiraApiConfig['types'] = {
         { fieldName: 'fieldContexts', fieldType: `list<${FIELD_CONTEXT_TYPE_NAME}>` },
       ],
       fieldsToHide: [
-        {
-          fieldName: 'id',
-        },
+        { fieldName: 'id' },
+        { fieldName: 'simplified' },
```
Comment (alonstern) [inline packages/jira-adapter/src/config/api_config.ts:842]: We don't really need to save it hidden on the instance. Can you delete it from the instance in fetch after you use it?

File: packages/jira-adapter/src/filters/issue_layout/issue_layout.ts [hunk 3]
```diff
@@ -204,20 +208,21 @@ const filter: FilterCreator = ({ client, config, fetchQuery, getElemIdFunc }) =>
...
-          setTypeDeploymentAnnotations(issueLayoutType)
-          await addAnnotationRecursively(issueLayoutType, CORE_ANNOTATIONS.CREATABLE)
-          await addAnnotationRecursively(issueLayoutType, CORE_ANNOTATIONS.UPDATABLE)
-          await addAnnotationRecursively(issueLayoutType, CORE_ANNOTATIONS.DELETABLE)
         }
-      })))
+        return undefined
+      })))).filter(isDefined)
+    elements.push(...issueLayouts)
+    await createReferences(config, issueLayouts, elements)
```
[the remark this answers is not in the document]
Comment (idoamit8) [reply packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:217]: create a new filter or move it to an existing different filter?

[the remark this answers is not in the document]
Comment (alonstern) [reply packages/jira-adapter/src/filters/issue_layout/issue_layout.ts:217]: A new filter

**Review by alonstern — APPROVED** [review]
👌

**Pull request merged**